providers/zohocrm/read.go

```diff
@@ -0,0 +1,55 @@
+package zohocrm
+
+import (
+	"context"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/urlbuilder"
+)
+
+// Read retrieves data based on the provided common.ReadParams configuration parameters.
+func (c *Connector) Read(ctx context.Context, config common.ReadParams) (*common.ReadResult, error) {
+	if err := config.ValidateParams(true); err != nil {
+		return nil, err
+	}
+
+	url, err := c.buildReadURL(config)
+	if err != nil {
+		return nil, err
+	}
+
+	modHeaders := constructHeader(config, url)
+
+	res, err := c.Client.Get(ctx, url.String(), modHeaders...)
+	if err != nil {
+		return nil, err
+	}
+
+	return common.ParseResult(res,
+		common.GetRecordsUnderJSONPath("data"),
+		getNextRecordsURL(url),
+		common.GetMarshaledData,
+		config.Fields,
+	)
+}
+
+func constructHeader(config common.ReadParams, url *urlbuilder.URL) []common.Header {
+	// Retrieve the since from the url.
+	since, ok := url.GetFirstQueryParam("since")
```

Comment:
In order for us to support since header, we need to preserve the value of the header(If-Modified-Since) used during the first call. This is bacause the next_page_token generated depends on that value.  

Sample Examples: 

First call Example:

Read(ctx, common.ReadParams{
    ObjectName: "Leads",
    Since: time.Now(-10*time.Hour)  // Lets say value is `2009-11-10 13:00:00 +0000`
    // Other Fields
})

The next_page_token generatedd after this call depends on the value passed in the Since field. So in the next calls we have to pass the same value. Changing this invalidates the token. 

Next Call Examples:

Read(ctx, common.ReadParams{
    ObjectName: "Leads",
    Since: "2009-11-10 13:00:00 +0000" // In all next-page requests this value should never change & is Required.
    NextPageToken: xxxxxxxxxxxxxxxxxxxx
})

So to preseve this value sent during the first request, I created a custom parameter (since). This param has nothing to do with Zoho. Am using this parameter to preserve the Since value used in the first request, And am returning it in the Next Page URL. So that whenever we make Next Page Requests. I can get it and Add the value in the header as well.

Reply:
Okay, so what you are saying that next_page_token alone does not work? and it only works if the `If-Modified-Since` is matching from previous call?

Reply:
That sounds really odd to me that Zoho does that. For sanity check, can you make a first call with `If-Modified-Since` header, then make 2 second calls as below and share screenshots of results? 
1. second page with `If-Modified-Since` 
2. second page without `If-Modified-Since`

I like the creative approach you are taking, but I am afraid that at some point, they might use or try to parse that param and return unexpected results. So just want to double check.

Reply:
Feel free to ping me directly on slack when you are done with this. We don't need to wait for another day to be re-reviewed.

Reply:
Header Tests:
First Call with the If-Modified-Since Header: 
Result:
<img width="1280" alt="Screenshot 2024-10-24 at 22 20 46" src="https://github.com/user-attachments/assets/cc0a1d37-7758-4d6a-8baf-cc9028eac9d4">

Second-Call with the `If-Modified-since` Header:
<img width="1280" alt="Screenshot 2024-10-24 at 22 22 02" src="https://github.com/user-attachments/assets/8b77259e-2b48-40e5-a16f-aaf4d529329d">

Second-Call  with no `If-Modified-since` Header, Responds with an error:
<img width="1280" alt="Screenshot 2024-10-24 at 22 22 54" src="https://github.com/user-attachments/assets/c0d175d5-8b97-4706-95c3-cd0d9a4fba1e">